lib/maze/server.rb

```diff
@@ -255,6 +254,10 @@ def reset!
         logs.clear
         invalid_requests.clear
       end
+
+      private
+
+      attr_writer :sampling_probability_generator
```

Comment:
I don't think this is used

```suggestion
```

Reply:
Without it my IDE told me that `@sampling_probability_generator` on L64 was undefined.  Perhaps there's a better way of defining it?

Reply:
This doesn't define `@sampling_probability_generator`, so I have no idea what your IDE is talking about or why it thinks this defines a class variable 😀

An `attr_writer` defines a setter method for the given symbol, that updates the instance variable of the same name. So this `attr_writer` expands to:

```ruby
def sampling_probability_generator=(value)
  @sampling_probability_generator = value
end
```

You only need it if you want to write `sampling_probability_generator = something` instead of `@sampling_probability_generator = something`, but you're using the instance variable everywhere (as you should!)

It's also totally valid to use instance variables without having given them a value before; they're just `nil`:

```ruby
class Abc
  class << self
    def x
      @abc
    end
    
    def y
      @abc = 1234
    end
  end
end

Abc.x # => nil
Abc.y # => 1234
Abc.x # => 1234
```

This definition doesn't do anything because it's never called and it'd be valid to use an instance variable before "defining" it anyway, so this sounds like a bug in your IDE

Reply:
I guess it's upset because you _could_ call `@sampling_probability_generator.next` before having set it, which would crash. I think we would want that to crash though, since that would be a bug in Maze Runner and should never happen as there's a default generator set automatically

Reply:
Actually, I've just removed it and the IDE is now fine.  It must either have been confused, or I genuinely got something wrong at that point of the refactor.